db/deals.go

```diff
@@ -76,6 +73,8 @@ func newDealAccessor(db *sql.DB, deal *types.ProviderDealState) *dealAccessor {
 			"Length":                &fieldDef{f: &deal.Length},
 			"Checkpoint":            &ckptFieldDef{f: &deal.Checkpoint},
 			"Error":                 &fieldDef{f: &deal.Err},
+			// Needed so the deal can be looked up by signed proposal cid
+			"SignedProposalCID": &signedPropFieldDef{prop: deal.ClientDealProposal},
```

Comment:
Please can we add a test for this field to the db/tests ?

Reply:
Added a test 👍